### PR TITLE
Split search command into separate file

### PR DESCRIPTION
**Pull request by Zopolis4**
Behaviour changes:
- Search both the name and the description of packages, rather than first searching the name and then only searching the description if no results are found.
- Don't list every package if nothing is passed.
- Don't abort if the search returns nothing.
- Fix the docopt string for `crew search`.
- Fix the help string for `crew search`.
  - The colors in help strings are still broken, which we should probably fix at some point.

Refactoring changes:
- Drop the now-unused `print_packages` and `list_packages` functions.
- Simplify the code for handling verbosity.
- Only loop over every package once.
- Only create the regex once.
- Drop the unnecessary error handling when loading packages.
  - Any issues of this sort will be caught by `prop_test` before they get merged.

Tested and working on `x86_64` and `i686`.

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/Zopolis4/chromebrew.git CREW_BRANCH=projectregula12 crew update
```

### CODE REVIEW

File: commands/help.rb [hunk 1]
```diff
@@ -101,16 +101,16 @@ def self.help(command)
       puts <<~EOT
         Look for package(s).
         Usage: crew search [-v|--verbose] [<pattern> ...]
-        If <pattern> is omitted, all packages will be returned.
```
Comment (uberhacker) [inline commands/help.rb:104]: I think we should keep this line to make it clear.

File: commands/help.rb [hunk 2]
```diff
@@ -101,16 +101,16 @@ def self.help(command)
...
+        Both the name and description of packages will be searched.
```
Comment (uberhacker) [inline commands/help.rb:104]: The only way anything is searched is if there is a pattern to filter the results.

File: commands/help.rb [hunk 3]
```diff
@@ -101,16 +101,16 @@ def self.help(command)
...
         If the package color is " + "green".lightgreen + ", it means the package is installed.
         If the package color is " + "red".lightred + ", it means the architecture is not supported.
+        If the package color is " + "blue".lightblue + ", it means the architecture is supported but the package is not installed.
         The <pattern> string can also contain regular expressions.
-        If `-v` or `--verbose` is present, homepage, version and license will be displayed.
+        If `-v` or `--verbose` is present, the homepage, version and license of found packages will be displayed.
         Examples:
-          crew search ^lib".lightblue + " will display all packages that start with `lib`.
-          crew search audio".lightblue + " will display all packages with `audio` in the name.
-          crew search | grep -i audio".lightblue + " will display all packages with `audio` in the name or description.
```
Comment (uberhacker) [inline commands/help.rb:112]: Why was this line removed?  It's definitely a good example.

Comment (Zopolis4) [reply commands/help.rb:112]: The same functionality can now be achieved with `crew search audio`.

File: commands/help.rb [hunk 4]
```diff
@@ -101,16 +101,16 @@ def self.help(command)
...
-          crew search git -v".lightblue + " will display packages with `git` in the name along with homepage, version and license.
+          crew search ^lib".lightblue + " will display all packages with a name or description that starts with `lib`.
+          crew search audio".lightblue + " will display all packages with `audio` in the name or description.
```
Comment (uberhacker) [inline commands/help.rb:112]: This is not true.   Compare the output of the commands below:
```
$ crew search audio
libaudiofile: The Audio File Library is a C-based library for reading and writing audio files in many common formats.
libfaudio: FAudio is an XAudio reimplementation that focuses solely on developing fully accurate DirectX Audio runtime libraries for the FNA project.
libspatialaudio: libspatialaudio is a library for Ambisonic encoding and decoding, filtering and binaural rendering.
pcaudiolib: PCaudiolib is a portable C audio library.
pulseaudio: PulseAudio is a sound system for POSIX OSes, meaning that it is a proxy for your sound applications.
rtaudio: RtAudio is a set of C++ classes that provide a common API (Application Programming Interface) for realtime audio input/output
webrtc_audio_processing: AudioProcessing library based on Googles implementation of WebRTC

$ crew search | grep -i audio | head -20
aalib: AA means Ascii Art - the AAlib (ascii art GFX library), BB (audiovisual demonstration for your terminal), aview (image browser/animation player), AAvga (SVGAlib wrapper for AA-lib), ttyquake (text mode quake), aa3d (random dot stereogram generator)...
alsa_lib: The Advanced Linux Sound Architecture (ALSA) provides audio and MIDI functionality to the Linux operating system.
audacious: Audacious is an open source audio player.
audacious_plugins: Audacious is an open source audio player.
audacity: Audacity is the world's most popular audio editing and recording app
cdparanoia: Compact Disc Digital Audio extraction tool
cdrkit: cdrkit is a suite of programs for recording CDs and DVDs, blanking CD-RW media, creating ISO-9660 filesystem images, extracting audio CD data, and more.
chromaprint: Chromaprint is a client-side library that implements a custom algorithm for extracting fingerprints from any audio source.
cras: ChromeOS Audio Server
ecasound: Ecasound is a software package designed for multitrack audio processing.
faac: Freeware Advanced Audio Coder
faad2: Freeware Advanced Audio (AAC) Decoder
ffmpeg: Complete solution to record, convert and stream audio and video
flac: FLAC stands for Free Lossless Audio Codec, an audio format similar to MP3, but lossless, meaning that audio is compressed in FLAC without any loss in quality.
gst_editing_services: GStreamer library for creating audio/video editors
jack: JACK (JACK Audio Connection Kit) refers to an API that provides a basic infrastructure for audio applications to communicate with each other and with audio hardware.
jack1: JACK (JACK Audio Connection Kit) refers to an API that provides a basic infrastructure for audio applications to communicate with each other and with audio hardware.
ladspa: Linux Audio Developer's Simple Plugin API
libaudiofile: The Audio File Library is a C-based library for reading and writing audio files in many common formats.
libavc1394: libavc1394 is a programming interface for the 1394 Trade Association AV/C (Audio/Video Control) Digital Interface Command Set.
```

File: commands/help.rb (hunk 2 above)
Comment (Zopolis4) [reply commands/help.rb:104]: Yes? I'm not really sure what you're commenting on here.

File: commands/help.rb (hunk 4 above)
Comment (Zopolis4) [reply commands/help.rb:112]: You are not running these commands on the updated branch. The behaviour has changed.

File: commands/help.rb (hunk 1 above)
Comment (Zopolis4) [reply commands/help.rb:104]: This line is no longer accurate.

File: commands/help.rb (hunk 4 above)
Comment (uberhacker) [reply commands/help.rb:112]: Oh wait, you changed the search algorithm.  Let me test this.

File: commands/search.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+require_relative '../lib/const'
+require_relative '../lib/package'
+require_relative '../lib/package_utils'
+
+class Command
+  def self.search(regex_string, verbose)
```
Comment (uberhacker) [inline commands/search.rb:6]: Why not just abort right away if regex_string is empty instead of looping through every package only to output nothing?

File: commands/help.rb (hunk 4 above)
Comment (uberhacker) [reply commands/help.rb:112]: Yeah, you were right.  I didn't update using that branch.  This works good.

File: commands/search.rb (hunk 5 above)
Comment (Zopolis4) [reply commands/search.rb:6]: docopt handles that for us:
image

Comment (uberhacker) [reply commands/search.rb:6]: Fair enough.  I'm sold.

**Review by uberhacker — APPROVED** [review]
